src/main/java/com/mongodb/hibernate/dialect/MongoDialect.java

```diff
@@ -91,4 +92,9 @@ public void contribute(TypeContributions typeContributions, ServiceRegistry serv
         typeContributions.contributeJavaType(ObjectIdJavaType.INSTANCE);
         typeContributions.contributeJdbcType(ObjectIdJdbcType.INSTANCE);
     }
+
+    @Override
+    public @Nullable String toQuotedIdentifier(@Nullable String name) {
+        return name;
```

Comment:
This single change makes all the new tests pass, and without it, they all fail. Nothing else related to quoting both in `Dialect` (including `Dialect.buildIdentifierHelper`) and in `DatabaseMetaData` had any observable effect on the tests added.

Reply:
It is a great news! Thanks for your PR. Withotu some minor changes it could be merged.